content/features/featuresDeepDive/mesh/boneInternals.md

```diff
@@ -0,0 +1,159 @@
+
+Note: the **local** matrices of the arm bones are not the same in both cases! For the left arm:
+* in the first image, rotation around the Z axis is 45°. As the **bind** matrix is the identity matrix, the rotation around Z is 0°, so the rotation around Z in the **final** matrix is 45-0=45°.
+* In the second image, the rotation around Z is 75°! This is because the **bind** matrix has a rotation of 30° around Z, so that the left arm bone matches the visual appearance of the mesh. So, to obtain a final rotation of 45° around Z, the rotation around Z in the **local** matrix must be 75°. So, after removing the **bind** matrix, the rotation around Z in the **final** matrix is 75-30=45°.
+
+You can see that the end result is the same in both cases (45° rotation around Z), which is why the final rendering is the same, but we achieve it in two different ways.
+
+Using a translation (0,0,0) for the **bind** matrix makes things even more difficult:
+
+| 45° rotation around Z | 45° rotation around Z and (1.2, 0.08, 0) translation |
+|-|-|
+| ![45° rotation around Z](/img/features/boneInternals/character_skeleton_no_bind_rotz45.jpg!400) | ![45° rotation around Z and (1.2, 0.08, 0) translation](/img/features/boneInternals/character_skeleton_no_bind_rotz45_transl.jpg!400) |
+
+<Playground id="#4IUBBK#6" title="No translation, no rotation for the bind matrix" description="Using a bind matrix with translation only"/>
+
+To obtain the same final rendering (for the left arm), we have to apply not only a 45° rotation around Z to the **local** bone matrix, but also a translation of (1.2, 0.08, 0)! This is considerably more difficult than above, and you can see that the bone positions/orientations are completely out of sync with the mesh.
+
+## Animation data generated by DCC tools
+
+DCC (Digital Content Creation) tools (such as Blender) generate animation data for bones in the form of matrices. These matrices are the **local** matrices of the bones, and as such also contain the transformation brought about by the **bind** matrix.
+
+For example, in the first playground where we've created the bones with a **bind** matrix that corresponds to the appearance of the mesh, if we create an animation with a single frame and this frame renders the mesh in its "rest" position, we'll export a rotation of 30° around Z for the left bone, and -30° for the right bone: we'll simply export the current **local** matrix of each bone. This is what Blender (or any other DCC tool) would do: for each frame of the animation, it would export the **local** matrix of each bone.
+
+This is compatible with our way of dealing with bones: when rendering, we remove the **bind** matrix from the **local** matrix (in fact, we do this calculation in world space, so we remove the **absolute bind** matrix from the **absolute** matrix), so in the example above, we'll render the mesh unmodified by the animation data.
```

Comment:
Great explanation! Just to confirm, does that mean that Blender and other DCC tools export both the local matrices and the bind matrix information for each bone?

Reply:
Yes, but it's not at the same place.

In the animation data (which is the purpose of this section), for each frame of animation, the local matrix of each bone is exported.

The bind matrix (actually, the inverse bind matrix) is also exported but in another location. For glTF, it is under the "skins" entry.